dbt/models/default/default.vw_pin_sale.sql

```diff
@@ -249,93 +244,166 @@ sales_val AS (
+            AS source,
+        md_sales.mydec_deed_type,
+        md_sales.sale_filter_ptax_flag,
+        md_sales.mydec_property_advertised,
+        md_sales.mydec_is_installment_contract_fulfilled,
+        md_sales.mydec_is_sale_between_related_individuals_or_corporate_affiliates, --noqa
+        md_sales.mydec_is_transfer_of_less_than_100_percent_interest,
+        md_sales.mydec_is_court_ordered_sale,
+        md_sales.mydec_is_sale_in_lieu_of_foreclosure,
+        md_sales.mydec_is_condemnation,
+        md_sales.mydec_is_short_sale,
+        md_sales.mydec_is_bank_reo_real_estate_owned,
+        md_sales.mydec_is_auction_sale,
+        md_sales.mydec_is_seller_buyer_a_relocation_company,
+        md_sales.mydec_is_seller_buyer_a_financial_institution_or_government_agency, --noqa
+        md_sales.mydec_is_buyer_a_real_estate_investment_trust,
+        md_sales.mydec_is_buyer_a_pension_fund,
+        md_sales.mydec_is_buyer_an_adjacent_property_owner,
+        md_sales.mydec_is_buyer_exercising_an_option_to_purchase,
+        md_sales.mydec_is_simultaneous_trade_of_property,
+        md_sales.mydec_is_sale_leaseback,
+        md_sales.mydec_is_homestead_exemption,
+        md_sales.mydec_homestead_exemption_general_alternative,
+        md_sales.mydec_homestead_exemption_senior_citizens,
+        md_sales.mydec_homestead_exemption_senior_citizens_assessment_freeze
+    FROM unique_sales AS uq_sales
+    FULL OUTER JOIN mydec_sales AS md_sales ON uq_sales.doc_no = md_sales.doc_no
+    LEFT JOIN town_class AS tc
+        ON COALESCE(uq_sales.pin, md_sales.pin) = tc.parid
+        AND COALESCE(uq_sales.year, md_sales.year) = tc.taxyr
 )
 
 SELECT
-    unique_sales.pin,
-    -- In the past, mydec sale dates were more precise than iasworld dates
-    -- which had been truncated
-    CASE
-        WHEN
-            mydec_sales.mydec_date IS NOT NULL
-            AND mydec_sales.mydec_date != unique_sales.sale_date
-            THEN mydec_sales.year_of_sale
-        ELSE unique_sales.year
-    END AS year,
-    unique_sales.township_code,
-    unique_sales.nbhd,
-    unique_sales.class,
-    -- In the past, mydec sale dates were more precise than iasworld dates
-    -- which had been truncated
-    CASE
-        WHEN
-            mydec_sales.mydec_date IS NOT NULL
-            AND mydec_sales.mydec_date != unique_sales.sale_date
-            THEN mydec_sales.mydec_date
-        ELSE unique_sales.sale_date
-    END AS sale_date,
-    -- From 2021 on iasWorld uses precise MyDec dates
-    COALESCE(
-        mydec_sales.mydec_date IS NOT NULL
-        OR YEAR(unique_sales.sale_date) >= 2021,
-        FALSE
+    combined_sales.pin,
+    combined_sales.year,
+    combined_sales.township_code,
+    combined_sales.nbhd,
+    combined_sales.class,
+    combined_sales.sale_date,
+    (
+        combined_sales.source = 'mydec'
+        OR YEAR(combined_sales.sale_date) >= 2021
```

Comment:
This logic might need to be a bit more complicated. We're favoring mydec for dates a bit differently than for other columns. Before 2021 we always favor mydec when it's available, and for 2021 and on we favor iasworld if it's available.